src/Workspaces/Core/Portable/EmbeddedLanguages/Common/EmbeddedSyntaxToken.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.Common
+{
+    internal struct EmbeddedSyntaxToken<TSyntaxKind> where TSyntaxKind : struct
+    {
+        public readonly TSyntaxKind Kind;
+        public readonly ImmutableArray<EmbeddedSyntaxTrivia<TSyntaxKind>> LeadingTrivia;
+        public readonly ImmutableArray<VirtualChar> VirtualChars;
+        public readonly ImmutableArray<EmbeddedSyntaxTrivia<TSyntaxKind>> TrailingTrivia;
+        internal readonly ImmutableArray<EmbeddedDiagnostic> Diagnostics;
+        public readonly object Value;
+
+        public EmbeddedSyntaxToken(
```

Comment:
From usage in subsequent PR, it seems that you never put default arrays for those parameters (but rather use empty arrays). Perhaps assert that immutable arrays in various types are not default?

Reply:
correct.  i will do that.

Reply:
#resolved.